### PR TITLE
Document "Tags from file names"

### PR DESCRIPTION
**Pull request by phw**
### Summary

This is a…

- [ ] Correction
- [x] Addition
- [ ] Restructuring
- [ ] Minor / simple change (like a typo)
- [ ] Other

### Reason for the Change
Adds documentation for the "Tags from file names" tool, see #31 .

### Description of the Change

Add documentation to "Other Picard Tasks" about using this tool

### Additional Action Required

- Maybe we should recreate the screenshots. I created them on my Linux system, but all other screenshots are for Windows (which as far as I can tell the majority of users use).
- I struggled with writing this up, took me several attempts. For some reason I found it especially hard to explain this. So please give some feedback for how to improve it
- I'd like to merge this to current master for 2.4.4. For 2.5 we should then likely extend this a bit and explain how to use underscore variables (`%_foo%`) to extract parts you don't want to match to actual tags or only want to use as variables. But this does not work in 2.4.4 yet as expected.

### CODE REVIEW

File: source/usage/tags_from_file_names.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+If your files for example consist of a track number and track title separated by a space
+(e.g. :file:`04 Heart of Gold.mp3`) you can use the matching pattern :samp:`%tracknumber% %title%`.
+Should the track number and title be separated by e.g. a colon like :file:`04 - Heart of Gold.mp3`
+the pattern needs to also include this separator, e.g. :samp:`%tracknumber% - %title%`.
+
+.. image:: ../images/tags_from_file_names_1.png
+   :width: 100 %
+
+Clicking on the "Preview" button next to the matching pattern will show a preview of the extracted
+tags for each file name.  Once you are satisfied with the result accept the changes with the
```
Comment (rdswift) [inline source/usage/tags_from_file_names.rst:35]:
```suggestion
tags for each file name.  Once you are satisfied with the result, you can accept the changes with the
```

File: source/usage/tags_from_file_names.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+(see :doc:`../usage/save`).
+
+
+Matching folders
+----------------
+
+The pattern can also match the parent folders of the file. To match for folders use a slash (``/``)
+as separator.  If for example the file is in a folder named after the album, which in turn is
+inside a folder named after the artist (i.e. :file:`Neil Young/Harvest/04 Heart of Gold.mp3`) you
+could match the artist, album, track number and title with a patter of
```
Comment (rdswift) [inline source/usage/tags_from_file_names.rst:47]:
```suggestion
could match the artist, album, track number and title with a pattern of
```

**Review by rdswift — COMMENTED** [review]
I think you've done a very good job of explaining this functionality.

I just had a thought...  As I understand it, this function is kind of like the file naming script in reverse.  Perhaps that concept could be used to help clarify if you think it might help.

**Comment by rdswift** [comment]
> * Maybe we should recreate the screenshots. I created them on my Linux system, but all other screenshots are for Windows (which as far as I can tell the majority of users use).

I can do that if you like, but I don't know that it matters much.  In fact, I was thinking that it might be good to have example screens from each of the platforms so that people don't get the impression that Picard is Windows only.  Alternately, we can use Windows based images, and just include a note somewhere in the introduction that "The images used in the documentation are based on an installation on Windows.  Other operating system images will be similar."

> * I struggled with writing this up, took me several attempts. For some reason I found it especially hard to explain this. So please give some feedback for how to improve it

I actually think you've done quite a good job of documenting the functionality.  I've been using Picard for years and never realized that this even existed.  :blush: 

> * I'd like to merge this to current master for 2.4.4. For 2.5 we should then likely extend this a bit and explain how to use underscore variables (`%_foo%`) to extract parts you don't want to match to actual tags or only want to use as variables. But this does not work in 2.4.4 yet as expected.

Sounds good to me.

File: source/usage/tags_from_file_names.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
+.. MusicBrainz Picard Documentation Project
+.. Prepared in 2020 by Bob Swift (bswift@rsds.ca)
```
[the remark this answers is not in the document]
Comment (phw) [reply source/usage/tags_from_file_names.rst:2]: Yes, it's a bit odd. I just copied the header over as most would probably. We had the same thing in Picard, where all the copyright headers for a long time just were copied over from one file to another and always showing the same names regardless of actual authorship. I'll just remove this line from the new file in this case.

File: source/usage/tags_from_file_names.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Sometimes files have no or bad tags, but the file names are well structured and follow a pattern.
+In this case you can use :menuselection:`"Tools --> Tags From &File Names..."` to generate the tags
+from the file names.
+
+
+Basic usage
+-----------
+
+To use this tool, select one or more files loaded into Picard and open the Tags From File Names
```
[the remark this answers is not in the document]
Comment (phw) [reply source/usage/tags_from_file_names.rst:17]: It really doesn't matter where they show up as long as they are in Picard. That means just in the file browser does not work, you need to drag them over. But whether the files are in unclustered, inside a cluster or matched to a release does not matter.

**Comment by phw** [comment]
@rdswift Thanks a lot for proof reading, I did the changes as suggested.

> In fact, I was thinking that it might be good to have example screens from each of the platforms so that people don't get the impression that Picard is Windows only.

You are right. It's a bit difficult with an application that shows differently on different platforms. Maybe we could just add additional screenshots of the main screen for multiple platforms on the Main Screen page. I think about leaving the current image as is, but at the bottom of the article just show a number of smaller pictures the user can click on to show the different appearance on major platforms. I will give this a try.

**Comment by phw** [comment]
> I've been using Picard for years and never realized that this even existed.

This seems to happen a lot. We have this functionality since the very first release of the Qt version of Picard. But people are always surprised learning that this exists. Sometimes people asking for this functionality are even told that this is not possible with Picard.

It also does not seem to be a much used feature in general. Neither are there many questions about how to get tags from file names nor where there many bug reports or improvement suggestions. That tells me that not too many people use this and for those who do it works reasonable well. But maybe more people would use it if they knew such a thing existed :)

**Review by rdswift — APPROVED** [review]

**Pull request merged**